pkg/controller/host/host_controller.go

```diff
@@ -1222,12 +1222,18 @@ func (r *ReconcileHost) ReconcileExistingHost(client *gophercloud.ServiceClient,
 	log.Info("current config is:", "values", current)
 
 	if instance.Status.Reconciled && r.StopAfterInSync() {
-		// Do not process any further changes once we have reached a
-		// synchronized state unless there is an annotation on the host.
 		if _, present := instance.Annotations[cloudManager.ReconcileAfterInSync]; !present {
-			msg := common.NoChangesAfterReconciled
-			r.NormalEvent(instance, common.ResourceUpdated, msg)
-			return common.NewChangeAfterInSync(msg)
+			if !host.IsUnlockedAvailable() {
```

Comment:
That scenario is tested. I'll prefer to monitor the host until unlocked/availbe before comparing the rest configurations, otherwise we may lost its state updates as we hit in the commit message